src/database/models.py

```diff
@@ -4379,5 +4381,185 @@ def simple_json(self):
 
     def full_json(self):
         return self.simple_json()
+    
+
+
+class CustomPage(BaseModel):
+    """
+        A class used to represent a custom page on the MassEnergize platform
+    """
+
+    title = models.CharField(max_length=LONG_STR_LEN, blank=True)
+    is_published = models.BooleanField(default=False, blank=True)
+    is_deleted = models.BooleanField(default=False, blank=True)
+    user = models.ForeignKey(UserProfile, on_delete=models.CASCADE, db_index=True)
+    slug = models.CharField(max_length=SHORT_STR_LEN, blank=True, null=True, unique=True)
```

Comment:
I think the slug should be unique for the community, but not necessarily unique for all custom pages.